impress-org/givewp-next-gen
# Refactor: form builder to use wp-scripts

## Description



This updates the form builder build process to use wp-scripts when loaded from the plugin.  This is necessary to offload dependency management to WordPress for use with custom external blocks.

## Affects



The form builder will not use the WordPress installation to load dependencies / gutenberg scripts.  Some imports that were coming directly from a package that is not publicly exposed were relocated and/or forked.

The main noticeable difference here is the secondary sidebar / block inserter.  It was too much of a rabbit hole to fork the components we were using so it is now simplified with the `<Library />` component that WP core and the isolated editor are using.

## Visuals



<img width="1440" alt="Screenshot 2023-05-10 at 11 56 50 AM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/1ad34248-5056-413f-91c9-002190ff7ae8">

<img width="1440" alt="Screenshot 2023-05-10 at 11 10 12 AM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/9bb674d5-29aa-4a32-845c-de872aea8ace">


## Testing Instructions



- Pull branch and run `npm install && npm run build'
- Open or create a next gen form and use the builder
- Everything should work the same except the block inserter/secondary sidebar which only shows child blocks when cursor is selecting section inner blocks in canvas.

## Pre-review Checklist
